src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/peripheral/ColonyPeripheral.java

```diff
@@ -0,0 +1,300 @@
+            worksData.add(MineColonies.workOrderToObject(workOrder));
+
+        return worksData;
+    }
+
+    @LuaFunction(mainThread = true)
+    public final Object getResearch() throws LuaException {
+        IColony colony = getColony();
+
+        IGlobalResearchTree globalTree = IGlobalResearchTree.getInstance();
+
+        Map<String, Object> result = new HashMap<>();
+        for (ResourceLocation branch : globalTree.getBranches()) {
+            try {
+                result.put(branch.toString(), MineColonies.getResearch(branch, globalTree.getPrimaryResearch(branch), colony));
+            } catch (CommandSyntaxException ex) {
+                AdvancedPeripherals.debug("Error getting research for branch " + branch + ": " + ex.getMessage(), org.apache.logging.log4j.Level.WARN);
+                ex.printStackTrace();
+            }
+        }
+
+        return result;
+    }
+
+    @LuaFunction(mainThread = true)
+    public final Object getWorkOrderResources(int id) throws LuaException {
+        IColony colony = getColony();
+
+        IWorkOrder workOrder = colony.getWorkManager().getWorkOrder(id);
+        if (workOrder == null) return null;
+
+        return MineColonies.builderResourcesToObject(colony, workOrder.getClaimedBy());
+    }
+
+    @LuaFunction(mainThread = true)
+    public final Object getBuilderResources(Map<?, ?> pos) throws LuaException {
+        IColony colony = getColony();
+
+        if (!(pos.containsKey("x") && pos.containsKey("y") && pos.containsKey("z")))
+            throw new LuaException("Coordinates expected");
+        BlockPos blockPos = LuaConverter.convertToBlockPos(pos);
+
+        return MineColonies.builderResourcesToObject(colony, blockPos);
+    }
+
+    @LuaFunction(mainThread = true)
+    public final Object getRequests() throws LuaException {
+        IColony colony = getColony();
+
+        IRequestManager requestManager = colony.getRequestManager();
+        //All requests assigned to players
+        IPlayerRequestResolver playerResolver = requestManager.getPlayerResolver();
+        //All requests not assigned to players
+        IRetryingRequestResolver requestResolver = requestManager.getRetryingRequestResolver();
+
+        Set<IToken<?>> tokens = new HashSet<>();
+        tokens.addAll(playerResolver.getAllAssignedRequests());
+        tokens.addAll(requestResolver.getAllAssignedRequests());
+
+        List<IRequest<?>> requests = new ArrayList<>();
+        for (IToken<?> token : tokens) {
+            IRequest<?> request = requestManager.getRequestForToken(token);
+            if (request.getRequest() instanceof IDeliverable)
+                requests.add(request);
+        }
+
+        List<Object> result = new ArrayList<>();
+        requests.forEach(request -> {
+            IDeliverable deliverableRequest = (IDeliverable) request.getRequest();
+            Map<String, Object> map = new HashMap<>();
+            map.put("id", request.getId().getIdentifier().toString());
```

Comment:
should this whole table construction exists in LuaConverter?

Reply:
Yeah I agree
But I don't think that fits to the scope of this PR currently, I will do that when I work on the colony integrator